analytics/cdk/emr-eks/lib/teams/emrEksTeam.ts

```diff
@@ -0,0 +1,130 @@
+import { Cluster, KubernetesManifest } from "aws-cdk-lib/aws-eks";
+import { ApplicationTeam, ClusterInfo, TeamProps } from "@aws-quickstart/eks-blueprints";
+import { readYamlDocument, loadYaml } from "@aws-quickstart/eks-blueprints/dist/utils"
+import { FederatedPrincipal, IManagedPolicy, IRole, ManagedPolicy, PolicyStatement, Role } from "aws-cdk-lib/aws-iam";
+import { Aws, CfnJson, CfnOutput, Stack } from "aws-cdk-lib";
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+import * as SimpleBase from 'simple-base';
+import { CfnVirtualCluster } from "aws-cdk-lib/aws-emrcontainers";
+
+
+export interface ExcutionRoleDefinition {
+  excutionRoleName: string,
+  excutionRoleIamPolicy?: IManagedPolicy
+  excutionRoleIamPolicyStatement?: PolicyStatement[],
+}
+
+export interface EmrEksTeamProps extends TeamProps {
+  virtualClusterNamespace: string,
+  createNamespace: boolean,
+  virtualClusterName: string,
+  excutionRoles: ExcutionRoleDefinition[]
+}
+
+
+export class EmrEksTeam extends ApplicationTeam {
+
+  private emrTeam: EmrEksTeamProps;
+
+  constructor(props: EmrEksTeamProps) {
+    super(props);
+    this.emrTeam = props;
+  }
+
+  setup(clusterInfo: ClusterInfo): void {
+    const cluster = clusterInfo.cluster;
+
+    const emrVcPrerequisit = this.setEmrContainersForNamespace(cluster, this.emrTeam.virtualClusterNamespace, this.emrTeam.createNamespace);
+
+    this.emrTeam.excutionRoles.forEach(excutionRole => {
+
+      const executionRolePolicy = excutionRole.excutionRoleIamPolicy ?
+        excutionRole.excutionRoleIamPolicy :
+        new ManagedPolicy(cluster.stack, `executionRole-${excutionRole.excutionRoleName}-Policy`, {
+          statements: excutionRole.excutionRoleIamPolicyStatement,
+        });
+
+      this.createExecutionRole(
+        cluster,
+        executionRolePolicy,
+        this.emrTeam.virtualClusterNamespace,
+        excutionRole.excutionRoleName);
+    });
+
+    const teamVC = new CfnVirtualCluster(cluster.stack, `${this.emrTeam.virtualClusterName}-VirtualCluster`, {
+      name: this.emrTeam.virtualClusterName,
+      containerProvider: {
+        id: cluster.clusterName,
+        type: 'EKS',
+        info: { eksInfo: { namespace: this.emrTeam.virtualClusterNamespace } },
+      },
+      tags: [{
+        key: 'created-with',
+        value: 'cdk-blueprint',
+      }],
+    });
+
+    teamVC.node.addDependency(emrVcPrerequisit)
+
+    new CfnOutput(cluster.stack, `${this.emrTeam.virtualClusterName}-id`, {
+      value: teamVC.attrId
+    })
+
+  }
+
+  private setEmrContainersForNamespace(cluster: Cluster, namespace: string, createNamespace: boolean): KubernetesManifest {
+
+    let emrContainersK8sRole = readYamlDocument(`${__dirname}/emrContainersRole.yaml`);
+    emrContainersK8sRole = emrContainersK8sRole.replace('<REPLACE-NAMESPACE>', namespace);
```

Comment:
It would be good to contextualize the namespace instead of adding `<REPLACE-NAMESPACE>` several times across the pattern.